### PR TITLE
Change how volume is handled

### PR DESCRIPTION
**Pull request by paulromano**
This pull request adds optional `volume` attributes to the `Cell`, `Material` and `Universe` classes. OpenMC doesn't use them directly other than for calculating microscopic MGXS over heterogeneous cells. This will allow volume information to be harvested by a downstream code, such as OpenDeplete. Additionally, I added a function to the Python API to run volume calculations directly (along with adding a new command-line flag to specify volume calculation mode).

@cjosey Let me know if this looks good to you.

FYI, I also noticed some mistakes in the materials.xml RELAX NG schema so I fixed those.

### CODE REVIEW

**Comment by cjosey** [comment]
Give me a moment.  I'm going to make an extension to opendeplete that leverages this API and I'll let you know how it goes.

**Comment by cjosey** [comment]
Alright, I've gone through it all and removed the ``volumes`` dictionary from my code.  It took longer than anticipated, as I did a refactor on a bunch of objects.  I'll put it up soon once I finally get rid of the need for a user to provide ``opendeplete.materials``.  It all works, and it's a lot cleaner.

One question I had is why one would run a volume calculation on materials and not cells.  The latter case, I imagine, would give far superior statistical performance with less complexity (aside from mapping the volume into the materials for computing total nuclide concentrations).   The second question is, would there be merit in providing ``executor.run_volume_calculation()`` with MPI information?

**Comment by paulromano** [comment]
@cjosey If one has a cell with distributed materials, you'd want the volume of each material, not the volume of the cell (which has multiple instances). You are right that providing `run_volume_calculation` with MPI would be beneficial -- I'll go ahead and add that.

**Comment by cjosey** [comment]
Ah, I had misinterpreted it.  So, if I were to perform a volume calculation on a distribmat case with a single cell and a thousand materials, each of those material volumes would be statistically independent despite describing the same quantity?

If that's the case, I will probably still code assuming volume of cells, but divide by the length of the distribmat array.

**Comment by paulromano** [comment]
That is correct -- each material volume would have its own independent value. In the most general case, each instance of the material doesn't necessary have to be the same volume, so in that case you'd have to get the volume of each. However, for most practical cases, as you point out you can get the volume of the cell and divide by the number of instances. When you say "I will ... code", is OpenDeplete still going to do something with volume, or is it expecting that the materials coming in from OpenMC have their volume attribute set properly?

**Comment by cjosey** [comment]
The PR I have open on opendeplete assumes that ``cell.volume`` is set to some value that corresponds to an individual cell volume, which I now know is incorrect.  I can either divide this by the length of ``cell.fill``, which would allow for low variance stochastic volumes (albeit in a confusing way), or I can switch to directly using ``material.volume``.

**Comment by paulromano** [comment]
I would recommend using `material.volume` directly so as not to assume that all distributed instances are the same volume. A user can still use a cell stochastic volume calculation and assign material volumes in a "smart" low-variance manner (we should probably have an example of this in a notebook).

**Comment by paulromano** [comment]
@cjosey Do you have any requested changes on this PR?

**Comment by cjosey** [comment]
Sorry, I was stuck in theory land the last few days.  I cannot think of any changes I would want, and I see no issues with the code.

**Comment by paulromano** [comment]
@wbinventor Mind giving a quick review?

**Comment by wbinventor** [comment]
Sure, I'll look through this later tonight.

File: openmc/executor.py [hunk 1]
```diff
@@ -51,8 +51,44 @@ def plot_geometry(output=True, openmc_exec='openmc', cwd='.'):
...
     """
+    if output:
+        output = 'full'
+    return _run([openmc_exec, '-p'], output, cwd)
 
-    return _run(openmc_exec + ' -p', output, cwd)
+
+def run_volume_calculation(threads=None, output=True, cwd='.',
+                           openmc_exec='openmc', mpi_args=None):
+    """Run stochastic volume calculations in OpenMC
```
Comment (wbinventor) [inline openmc/executor.py]: Perhaps this is just my naiveté shining through, but does this method just run and calculate the volume for all cells in a simulation? I presume one must have first created a "volume.xml" input file? If so, can this information be added to the docstring?

File: openmc/material.py [hunk 2]
```diff
@@ -82,6 +86,8 @@ def __init__(self, material_id=None, name='', temperature=None):
         self._density = None
         self._density_units = ''
         self._depletable = False
+        self._volume = None
+        self._atoms = {}
```
Comment (wbinventor) [inline openmc/material.py]: Does `_atoms` not deserve a property and mention in the class docstring?

Comment (wbinventor) [inline openmc/material.py]: I suppose it is only used the `get_nuclide_densities()` method.

File: openmc/universe.py [hunk 3]
```diff
@@ -118,6 +128,12 @@ def name(self, name):
         else:
             self._name = ''
 
+    @volume.setter
+    def volume(self, volume):
+        if volume is not None:
+            cv.check_type('universe volume', volume, Real)
```
Comment (wbinventor) [inline openmc/universe.py]: I don't think that it is standard convention to use the class name in the type check methods for the rest of our properties. Do you think that this is what we should move towards?

File: openmc/executor.py (hunk 1 above)
Comment (paulromano) [inline openmc/executor.py]: That is indeed what it does; I'll beef up the docstring to explain better.

File: docs/source/pythonapi/index.rst [hunk 4]
```diff
@@ -185,6 +185,7 @@ Running OpenMC
    :template: myfunction.rst
 
    openmc.run
+   openmc.run_volume_calculation
```
Comment (paulromano) [inline docs/source/pythonapi/index.rst]: I'm ok with that. I chose "run" as some kind of indication that the Fortran code is being run, but "plot_geometry" already breaks that convention.

File: openmc/material.py (hunk 2 above)
Comment (paulromano) [inline openmc/material.py]: Yeah, I kind of debated this one. I'm not 100% sure I want an atoms property, so for now I'm holding off on exposing it. As you point out, it's only used internally right now.

File: openmc/universe.py (hunk 3 above)
Comment (paulromano) [inline openmc/universe.py]: Perhaps- I don't have a strong opinion on the matter. I was just following the other setters for the universe class. Probably worth revisiting this for the API as a whole at some point.

**Comment by wbinventor** [comment]
Reopening this PR to trigger Travis to run again since it failed for Python 2 earlier for no apparent reason.

**Pull request merged**